### ISSUE TITLE
AutoMagical tiny URL

### ISSUE THREAD

**Issue by pahosler** [campDevs/DiscordBot#29]
Sometimes links we find and want to post in channel have a URL that is larger than the character limit in discord. Pass your long URLs to the bot, and it will send a discord friendly URL to the channel.

**Comment by pahosler** [comment]
one problem, it is possible that discord blocks character length before sending anywhere, so this may not work

**Comment by jodylecompte** [comment]
Links are definitely a bit of a pain point in general for Discord, I think. One example that comes to mind is I wanted to be able to change the text of a link, but apparently Discord does not allow this per their core team because it allows people to hide unsafe URLs and people might click them.

On Gitter, this would have made sense and we had multiple occasions I remember of fake URLs being posted; but Discord does the whole "Whoops, are you sure you wanna go here? <link>" so I'm not sure why the paranoid reaction to URLs.

**Comment by jodylecompte** [comment]
@pahosler I wasn't able to find any more information on how we could potentially implement this. Do you have any ideas?

**Comment by jodylecompte** [comment]
I'm going to go ahead and close this as a second look around did not reveal any way to work around this. If anyone has a theory or knows of a solution, we can re-open and discuss.

**Comment by joker314** [comment]
What do you mean? You just use a URL shortener? We could even make our own. You could send this in a DM:

> **Me**: -url https://firstpart
> **Bot**: Received
> **Me**: -url .com/secondpart
> **Bot**: Received
> **Me**: -url
> **Bot**: Copy the below URL:
```
freecodecampbot.glitch.me/3
```

**Comment by joker314** [comment]
To be honest, I prefer TinyURL, as it has an API: `http://tinyurl.com/api-create.php?url=http://yourdomain.com/with/a/very/long/url`. You feed the bot a URL in chunks, and it does it for you.

Sometimes, you want to embed a URL in a really long message. If that happens, when you DM the bot, it will fit in one message, which won't be too inconvenient! Thoughts?

**Comment by jodylecompte** [comment]
@joker314 The goal wasn't just to provide a URL shortener but to work around the max URL length in Discord. So instead of getting an error from Discord, the bot would just post a short URL.

The problem is that in order for bot to be able to receive the message, Discord first has to be able to accept it. The only work around that occured to me was a webhook / form of sorts where we could share a link and the bot share a short URL in its stead -- but with that level of manual work neeeded URLs could just be put in google shortener or equivilant.

**Comment by joker314** [comment]
Indeed. The maximum length for embeds (iirc) is longer than that of regular messages. Hence, we could do a

```
+shortenurl start
https://
exa
mple
.com
+shortenurl post #general
```

(w/ ~1m timeouts)

The bot would then post an embed containing the concatenated link, as well as the user ID of the person who made the request. The bot would verify that the user is not muted before processing the request.

Thoughts?
